athletics.lic

```diff
@@ -0,0 +1,223 @@
+    walk_to(@settings.safe_room)
+  end
+  
+  def climb?(room, targets)
+    targets.each do |target|
+      walk_to(room)
+      return true if DRRoom.npcs.length >= 3
+
+      fput "climb #{target}"
+      pause
+      waitrt?
+      return false if DRSkill.getxp('Athletics') >= @end_skill
+    end
+    true
+  end
+  
+  def play_song?(blocking = false)
+    UserVars.song = @song_list.first.first unless UserVars.song
+    @did_play = true
+    case bput("play #{UserVars.song}", 'dirtiness may affect your performance', 'slightest hint of difficulty', 'You begin a', 'You struggle to begin', 'You\'re already playing a song', 'You effortlessly begin', 'You begin some', 'You cannot play', 'Play on what instrument', 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play')
+    when 'Play on what instrument'
+      @no_instrument = true
+      return false
+    when 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play'
+      return true
+    when 'dirtiness may affect your performance'
+      if DRSkill.getrank('Performance') < 20
+        echo "Skipping cleaning of zills due to low rank of Performance: #{DRSkill.getrank('Performance')}"
+        return true
+      end
+      stop_play
+      clean_zills
+      return play_song?
+    when 'You begin a', 'You effortlessly begin', 'You begin some'
+      stop_play
+      UserVars.song = @song_list[UserVars.song] || @song_list.first.first
+      return play_song?
+    when 'You struggle to begin'
+      if UserVars.song != @song_list.first.first
+        stop_play
+        UserVars.song = @song_list.first.first
+        return play_song?
+      end
+    end
+ 
+    return true unless blocking
+ 
+    Flags.reset('song')
+    pause 1 until Flags['song']
+    true
+  end
+ 
+  def clean_zills
+    cloth = @settings.cleaning_cloth
+    bput('remove my zills', 'You slide')
+    bput("get my #{cloth}", 'You get')
+ 
+    loop do
+      case bput("wipe my zills with my #{cloth}", 'Roundtime', 'not in need of drying', 'You should be sitting up')
+      when 'not in need of drying'
+        break
+      when 'You should be sitting up'
+        fix_standing
+        next
+      end
+      pause 1
+      waitrt?
+ 
+      until /you wring a dry/i =~ bput("wring my #{cloth}", 'You wring a dry', 'You wring out')
+        pause 1
+        waitrt?
+      end
+    end
+ 
+    until /not in need of cleaning/i =~ bput("clean my zills with my #{cloth}", 'Roundtime', 'not in need of cleaning')
+      pause 1
+      waitrt?
+    end
+ 
+    bput('wear my zills', 'You slide')
+    bput("stow my #{cloth}", 'You put')
+  end
+ 
+  def stop_play
+    return unless @did_play
+    return if @no_instrument
+    @did_play = false
+    bput('stop play', 'You stop playing your song', 'In the name of', "But you're not performing")
+    Flags['song'] = true
+  end
+  
+end
+before_dying do
+  DRC.bput('stop climb','.*')
+end
```

Comment:
Is there a specific case this guards against? I try to avoid before_dying where I can.